ibc_public/utils_contrasts.py

```diff
@@ -278,10 +280,143 @@ def multi_modal(design_matrix_columns):
+               'objects_geometry-objects_others',
+               'objects_edge-objects_others',
+               'objects_photo-objects_others',
+               'places_geometry-places_others',
+               'places_edge-places_others',
+               'places_photo-places_others'
+    ]
+    contrast_names = localizer_ + humanbody_ + animals_ + faces_ +\
+                     flora_ + objects_ + places_ + others_
+    if design_matrix_columns is None:
+        return dict([(name, []) for name in contrast_names])
+    con = _elementary_contrasts(design_matrix_columns)
+    contrasts = dict([(name, con[name]) for name in localizer_ +\
+                      humanbody_ + animals_ + faces_ + flora_ +\
+                      objects_ + places_])
+    localizer = np.sum([con[x] for x in localizer_], 0)
+    humanbody = np.sum([con[x] for x in humanbody_], 0)
+    animals = np.sum([con[x] for x in animals_], 0)
+    faces = np.sum([con[x] for x in faces_], 0)
+    flora = np.sum([con[x] for x in flora_], 0)
+    objects = np.sum([con[x] for x in objects_], 0)
+    places = np.sum([con[x] for x in places_], 0)
+    allstim_ =  faces_ + humanbody_ + animals_ + flora_ + objects_  +\
+        places_
+    geometry_ = ['humanbody_geometry', 'animals_geometry', 'faces_geometry',
+                 'flora_geometry','objects_geometry', 'places_geometry']
+    edge_ = ['humanbody_edge', 'animals_edge', 'faces_edge',
+                 'flora_edge','objects_edge', 'places_edge']
+    photo_ = ['humanbody_photo', 'animals_photo', 'faces_photo',
+                 'flora_photo','objects_photo', 'places_photo']
+    allstim = np.sum([con[x] for x in allstim_], 0)
+    geometry = np.sum([con[x] for x in geometry_], 0)
+    edge = np.sum([con[x] for x in edge_], 0)
+    photo = np.sum([con[x] for x in photo_], 0)
+    contrasts['localizer_faces-others'] = 7 * con['localizer_faces'] -\
+        localizer
+    contrasts['localizer_humanbody-others'] = 7 *\
+        con['localizer_humanbody'] - localizer
+    contrasts['localizer_words-others'] = 7 * con['localizer_words'] -\
+        localizer
+    contrasts['localizer_nonsensewords-others'] = 7 *\
+        con['localizer_nonsensewords'] - localizer
+    contrasts['localizer_numbers-others'] = 7 * con['localizer_numbers'] -\
+        localizer
+    contrasts['localizer_places-others'] = 7 * con['localizer_places'] -\
+        localizer
+    contrasts['localizer_objects-others'] = 7 * con['localizer_objects'] -\
+        localizer
+    contrasts['localizer_checkerboards-others'] = 7 *\
+        con['localizer_checkerboards'] - localizer
+    contrasts['humanbody-other'] = 5 * humanbody - allstim
+    contrasts['animals-other'] = 5 * animals - allstim
+    contrasts['faces-other'] = 5 * faces- allstim
+    contrasts['flora-other'] = 5 * flora - allstim
+    contrasts['places-other'] = 5 * places - allstim
+    contrasts['objects-other'] = 5 * objects - allstim
+    contrasts['geometry-others'] = 2 * geometry - allstim
```

Comment:
```suggestion
    contrasts['geometry-others'] = 3 * geometry - allstim
``` as we want the contrasts to sum to 0 ?

Reply:
Same thing for the following contrasts.